docs/developers/design/001_linear_spinwave_theory.md

```diff
@@ -0,0 +1,95 @@
+# The Maths behind Linear Spin Wave Theory Calculations
+
+This documentation describes the maths behind linear spin wave theory calculations as implemented in the (Matlab) [SpinW](https://github.com/spinw/spinw) code.
+A fuller description can be found in the papers of [S. Petit (2011)](https://doi.org/10.1051/sfn/201112006) and [Toth and Lake (2015)](https://doi.org/10.1088/0953-8984/27/16/166002).
+(We shall mostly be using the notation of Petit as this is what is used in the SpinW Matlab code).
+
+
+## Introduction
+
+Spin waves are collective excitations (normal modes) of a lattice of atomic spins
+(atoms with unfilled electronic shells and hence a net magnetic moment) coupled by exchange interactions.
+These excitations can be described using a semiclassical approach using an equation of motion in which
+the internal magnetic field generated by the moment on one atom causes a torque on its neighbour. [citation needed]
+The resulting wave is a precession of the spins about their ordered direction, with a net phase between site.
+Another way of thinking about these excitations is as (quantised) quasiparticles in a field theory.
+This is "linear spin wave theory" and its quasiparticles are "magnons".
+
+
+## The Holstein-Primakoff transformation
+
+Consider a single atom with total spin quantum number $S$, which can be in states labelled by $`S_z=-S,-S+1,...,S-1,S`$.
+In the ordered state, this spin is in the state with maximum $`S_z=S`$.
+A small deviation from this (i.e. initiating a spin wave) will change the state to $`S_z=S-1`$.
+This can be described by the spin lowering operator $`\hat{S}^-`$.
+Likewise, restoring the ordered state from this deviation is described by the spin raising operator $`\hat{S}^+`$.
+
+Now, the _Holstein-Primakoff_ transformation is a mapping between these lowering and raising operators
+to bosonic creation $`\hat{b}^{\dagger}`$ and annihilation $`\hat{b}`$ operators as follows (we drop the hats):
+
+```math
```

Comment:
Ah sorry the $`\sqrt{2S}`$ terms refer to the scalar "spin length" or total spin quantum number $S$ rather than the vector $`\mathbf{S}=(S^x, S^y, S^z)`$. I'll make that clear in the text.

Reply:
@lucas-wilkins Ah, I get your meaning now - it's the $`S+ = b\sqrt{2S - b^{\dagger}}`$ term you mean.

Right, so in our particular case we only ever calculate the _square_ of these operators in the $`\mathbf{S}_i \mathrm{J}_{ij} \mathbf{S}_j`$ and $`\mathbf{S}_i \mathrm{A}_{i} \mathbf{S}_i`$ terms and in the Zeeman term we take the field to be along $z$ and use only the $S^z$ operator.

If we ever need to calculate the matrix elements of odd powers of these operators, then we would need to perform a Taylor expansion of the square root - which is tricky and requires partial differentiation of the operators. However, the authors [of this paper](https://doi.org/10.1103/PhysRevResearch.2.043243) worked out a way around this.